### ISSUE TITLE
Selection toolbar: bug in behaviour of selection start/end

### ISSUE THREAD

**Issue by DavidBailes** [audacity/audacity#4471]
### Steps to reproduce

With default configuration:
1. Open Audacity.
2. Click on the seconds digit of selection start, so that it's the focus.
3. Press up arrow
4. Press up arrow

Alternative scenario:
1. Open Audacity.
2. Click on the seconds digit of selection start, so that it's the focus.
3. Enter "1" using the keyboard
4. Click on the seconds digit of selection start and enter "2" using the keyboard

### Expected behavior

Results of steps:
3. digit changes to 1, and corresponding digit in selection end changes to 1
4. digit changes to 2, and corresponding digit in selection end changes to 2

### Actual behavior

Results of steps:
3. digit remains 0, but the corresponding digit in selection end changes to 1.
4. digit changes to 1, corresponding digit in selection end remains 1.

### Audacity Version

other
https://github.com/audacity/audacity/actions/runs/4561247541

### Operating system

Windows, macOS

### Additional context

This effect can be seen using any digit of any format.
Same effect can be seen when selection start and end are the same, and digits in selection end are decremented using down arrow.

**Comment by DavidBailes** [comment]
> Testing on W10 with latest alpha: audacity-win-3.3.0-alpha-20230329+8715a71-x64-msvc2022
> 
> This works fine for me:

As given in the Audacity version section of the bug report, this bug occurs in crib's bottom bar ui branch, where the lower tool dock looks something like this:
![image](https://user-images.githubusercontent.com/11887021/228798589-113e6b68-5f1b-4bfa-ae24-1554ac9872ed.png)

**Comment by petersampsonaudacity** [comment]
OOH yest there is an oddity here with @crsib  Dmitry's branch build with the updated bottom tooldock and trimmed Selection Toolbar  - a regression in fact:
![image](https://user-images.githubusercontent.com/50205984/228800974-d1a62fbc-cba2-443f-9d3b-42262c7c9afe.png)

**Comment by dozzzzer** [comment]
I could reproduce that on macOS as well

**Comment by dozzzzer** [comment]
Assigning **P2** since it doesn't only happen when pressing arrow keys but also when entering values using the keyboard.

**Comment by crsib** [comment]
It will be fixed in the following few minutes anyway

**Comment by crsib** [comment]
Should be fixed by #4414

**Comment by DavidBailes** [comment]
> Same effect can be seen when selection start and end are the same, and digits in selection end are decremented using down arrow.

For this case, after https://github.com/audacity/audacity/pull/4414 when digits in selection end are decremented using down arrow, selection start incorrectly is not changed.

**Comment by crsib** [comment]
Hmm, well, that is an easy fix.

**Comment by dozzzzer** [comment]
Fixed by #4414
